app/models/scraper.rb

```diff
@@ -114,6 +114,22 @@ def latest_successful_run_time
     latest_successful_run.finished_at if latest_successful_run
   end
 
+  def last_run_finished_successfully?
+    if last_run
+      last_run.finished_successfully?
+    else
```

Comment:
You could make this method much more concise by changing it to just

```ruby
last_run && last_run.finished_successfully?
```

And same with the next method.